mantidproject/mantid
# Move the AlgorithmProgress widget to C++

**Description of work.**
The algorithm progress widget is now in C++.

**To test:**
- Run a script that runs many small algorithms (thus gets to update what the progress bar is looking after quickly)
- Run that script simultaneously in multiple code editors
- Run a script that is slower and has multiple progress stages
- Open the details window, and repeat the above

The progress bar should also show progress messages sent from the algs when reporting.



Fixes #25063 
Fixes #23582



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
